test/bprotocol/bprotocol-lusd-hodl.js

```diff
@@ -0,0 +1,233 @@
+
+        lqtyPrice = new BigNumber(await priceOracle.getPrice(lqty));
+        console.log("LQTY price:", lqtyPrice.toFixed()/D18.toFixed());
+
+        oldValue = (farmerOldBalance).plus((oldPotPoolBalance.times(oldHodlSharePrice).times(lqtyPrice)).div(1e36));
+        newValue = (farmerOldBalance).plus((newPotPoolBalance.times(newHodlSharePrice).times(lqtyPrice)).div(1e36));
+
+        console.log("old value: ", oldValue.toFixed()/D18.toFixed());
+        console.log("new value: ", newValue.toFixed()/D18.toFixed());
+        console.log("growth: ", newValue.toFixed() / oldValue.toFixed());
+
+        apr = (newValue.toFixed()/oldValue.toFixed()-1)*(24/(blocksPerHour/272))*365;
+        apy = ((newValue.toFixed()/oldValue.toFixed()-1)*(24/(blocksPerHour/272))+1)**365;
+
+        console.log("instant APR:", apr*100, "%");
+        console.log("instant APY:", (apy-1)*100, "%");
+
+        console.log("fLQTY in potpool: ", newPotPoolBalance.toFixed()/D18.toFixed());
+      
+        await Utils.advanceNBlock(blocksPerHour);
+      }
+      // withdrawAll to make sure no doHardwork is called when we do withdraw later.
+      await vault.withdrawAll({ from: governance });
+
+      // wait until all reward can be claimed by the farmer
+      await Utils.waitTime(86400 * 30 * 1000);
+      await potPool.exit({from: farmer1});
+      await vault.withdraw(new BigNumber(await vault.balanceOf(farmer1)).toFixed(), { from: farmer1 });
+      
+      console.log("-----------------------------");
+      console.log("-------DONE LOOPING --------");
+      console.log("-----------------------------");
+
+      let farmerNewBalance = new BigNumber(await underlying.balanceOf(farmer1));
+      console.log('farmerOldBalance LUSD', farmerOldBalance.toFixed()/D18.toFixed());
+      console.log('farmerNewBalance LUSD', farmerNewBalance.toFixed()/D18.toFixed());
+      let farmerNewFLqty = new BigNumber(await hodlVault.balanceOf(farmer1));
+
+      Utils.assertBNGt(farmerNewFLqty, farmerOldFLqty);
+
+      console.log("-----------------------------");
+
+      console.log("Farmer had fLQTY before", farmerOldFLqty.toFixed()/D18.toFixed());
+      console.log("Farmer has fLQTY after ", farmerNewFLqty.toFixed()/D18.toFixed());
+
+      console.log("-----------------------------");
+
+      oldValue = farmerOldBalance;
+      newValue = (farmerOldBalance.times(newSharePrice)).div(D18.toFixed()).plus((farmerNewFLqty.times(newHodlSharePrice).times(lqtyPrice)).div(1e36));
+
+      console.log("Farmer total value before", oldValue.toFixed()/D18.toFixed());
+      console.log("Farmer total value after:", newValue.toFixed()/D18.toFixed());
+      console.log("Farmer total growth: ", newValue.toFixed() / oldValue.toFixed());
+      
+      console.log("-----------------------------");
+
+      Utils.assertBNGt(newValue, oldValue);
+
+      apr = (newValue.toFixed()/oldValue.toFixed()-1)*(24/(blocksPerHour/272))*365;
```

Comment:
Not really important, but for overall apr/apy multiply `blocksPerHour` with `hours`

Reply:
Thank you. fixed and updated the PR description @CryptJS13